source/common/stats/thread_local_store.h

```diff
@@ -122,6 +123,7 @@ class ThreadLocalStoreImpl : public StoreRoot {
   ScopePtr default_scope_;
   std::list<std::reference_wrapper<Sink>> timer_sinks_;
   const std::vector<TagExtractorPtr>* tag_extractors_{};
+  const std::vector<Tag>* default_tags_{};
```

Comment:
Adding a class as mentioned in 1 above would mean that this class doesn't have to keep track of two separate sets of related data and know how to process it.